internal/controller/olsconfig_controller.go

```diff
@@ -159,7 +163,12 @@ func (r *OLSConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	// Update status condition for Custom Resource
 	r.updateStatusCondition(ctx, olsconfig, typeCRReconciled, true, "Custom resource successfully reconciled", nil)
 
-	return ctrl.Result{}, nil
+	// Requeue if no reconciliation is scheduled in future.
+	if r.NextReconcileTime.After(time.Now()) {
+		return ctrl.Result{}, nil
+	}
+	r.NextReconcileTime = time.Now().Add(r.Options.ReconcileInterval)
+	return ctrl.Result{RequeueAfter: r.Options.ReconcileInterval}, nil
```

Comment:
so we're requeuing the event despite having processed it successfully, just to trigger a future reconciliation?

Reply:
Correct. Just keep the periodic reconcile going - only for a case if  we missed a event.